### ISSUE TITLE
[feature request] Add option to for destination environment in copy_content()

### ISSUE THREAD

**Issue by rhjostone** [release-engineering/pubtools-pulplib#214]
#### Summary

With how many pulp environments there can be, content units can easily fall out of sync between them. It doesn't appear that [`copy_content()`](https://release-engineering.github.io/pubtools-pulplib/api/client.html#pubtools.pulplib.Client.copy_content) has a way of setting a destination environment to complement the `to_repository` parameter. Adding such a parameter would drastically simplify content synchronization.

#### Issue Type
- Feature Request

#### Current behavior

There doesn't appear to be a built-in way to synchronize content units between pulp environments when using `copy_content()`.

#### Desired behavior

`copy_content()` should have ideally have an optional parameter for a destination environment, such as `client.copy_content(TARGET_REPO, DESTINATION_REPO, to_client=Client(url=DEST_URL, cert=CERT))`

#### Use case

A `stage` pulp environment is occasionally used for comparing content against `prod`, however even when running `pub` to push a compose and update pulp, the packages in the `stage` pulp environment aren't guaranteed to be 1:1 with prod. Copying pulp content units directly between environments may be more reliable.

**Comment by rohanpm** [comment]
I don't see a reasonable way how this could be implemented.  Copying units between one Pulp server and another isn't something that the Pulp API supports.

The closest analogy is [sync](https://release-engineering.github.io/pubtools-pulplib/api/pulpcore.html#pubtools.pulplib.Repository.sync), which roughly does (for example) asking Pulp to download a remote yum repo and import all content found within.

**Comment by rhjostone** [comment]
@rohanpm sync sounds interesting. Is there a basic usage example I could follow?

**Comment by rohanpm** [comment]
Not really, it was added for usage with docker-pulp repos, and that service has now been retired.

For other types of repos I believe it falls into the category of: theoretically should work, but since nobody uses it, bugs are likely.
